system/lib/wasmfs/wasmfs.cpp

```diff
@@ -66,6 +67,30 @@ extern "C" void wasmfs_flush(void) {
   // Flush our own streams. TODO: flush all backends.
   (void)SpecialFiles::getStdout()->locked().flush();
   (void)SpecialFiles::getStderr()->locked().flush();
+
+  std::stack<std::shared_ptr<Directory>> toFlush;
+  toFlush.push(wasmFS.getRootDirectory());
+
+  while(!toFlush.empty()) {
+    auto dir = toFlush.top();
+    toFlush.pop();
+    
+    auto lockedDir = dir->locked();
+    int nentries = lockedDir.getNumEntries();
+    Directory::MaybeEntries entries = lockedDir.getEntries();
+    for (int i = 0; i < nentries; i++) {
+      auto entry = entries->at(i);
```

Comment:
Can this not be
```suggestion
    for (auto& entry : *lockedDir.getEntries()) {
```
Though, regardless of this, we should check if `getEntries()` returns an error (using `getError()`).

Reply:
That makes sense. Out of curiosity, is there any reason that `wasmfs_readdir_get` is structured without using a for-each loop? (I used that code as a starting point for this section)

Reply:
That function doesn't have a loop at all, I see,

https://github.com/emscripten-core/emscripten/blob/f4398806c0afa2b8dc50a0a73d57e38588b77223/system/lib/wasmfs/js_api.cpp#L331-L336

I think that's because of how the readdir API is built, that you can call into it over time, so it can't loop all at once?